src/libtexture/imagecache.cpp

```diff
@@ -791,36 +794,41 @@ ImageCacheFile::init_from_spec()
 
 
 bool
-ImageCacheFile::read_tile(ImageCachePerThreadInfo* thread_info, int subimage,
-                          int miplevel, int x, int y, int z, int chbegin,
-                          int chend, TypeDesc format, void* data)
+ImageCacheFile::read_tile(ImageCachePerThreadInfo* thread_info,
+                          const TileID& id, void* data)
```

Comment:
Wow, I MUCH prefer the parsing of the structure here, that's WAY nicer as an API.